unleash-yggdrasil/src/strategy_upgrade.rs

```diff
@@ -18,6 +18,28 @@ pub fn upgrade(strategies: &Vec<Strategy>, segment_map: &HashMap<i32, Segment>)
     rule_text
 }
 
+pub fn build_variant_rules(
```

Comment:
This is the magic that makes this work - we build a set of rules for each individual strategy and return a tuple of [rule, Vec<Variant>]

Reply:
We're only doing this for strategies with variants https://github.com/Unleash/yggdrasil/pull/41/files#diff-fcd282bba31d2f9abf99d8187bc54b14f83a5863c9c6dcc9302204af1d7b8abbR27

I'm wondering what would happen if you have some strategies with variants and some without...

Reply:
Won't matter, all strategies will go through this flow, if they have variants, they'll get a rule for that, if they don't, they'll get none

When we check the variant, we first check if the strategy has a compiled variant set, if it does we'll use that, if it doesn't we'll fallback to the "normal" variant or none if those also don't exist